File: arcgis_map_android/android/src/main/kotlin/esri/arcgis/flutter_plugin/model/AnimationOptions.kt
package esri.arcgis.flutter_plugin.model

enum class AnimationCurve {
    LINEAR,
    EASE,
    EASY_IN,
    EASY_OUT,
    EASY_IN_OUT,
}

data class AnimationOptions(
    val duration: Double,
    val animationCurve: AnimationCurve,
)

File: arcgis_map_ios/ios/Classes/Models/AnimationOptions.swift
//
//  AnimationOptions.swift
//  arcgis_map_ios
//
//  Created by Julian Bissekkou on 01.12.22.
//

import Foundation
import ArcGIS

struct AnimationOptions: Codable {
    var duration: Double?
    var animationCurve: String
}

extension AnimationOptions {
    func arcgisAnimationCurve() -> AGSAnimationCurve {
        switch animationCurve {
        case "linear":
            return .linear
        case "ease-in":
            return .easeInCubic
        case "ease-out":
            return .easeOutCubic
        case "ease-in-out":
            return .easeInOutCubic
        default:
            return .linear
        }
    }
}
